clientconn.go

```diff
@@ -673,21 +673,31 @@ func (cc *ClientConn) handleResolvedAddrs(addrs []resolver.Address, err error) {
 	if cc.dopts.balancerBuilder != nil && cc.balancerWrapper == nil {
 		cc.balancerWrapper = newCCBalancerWrapper(cc, cc.dopts.balancerBuilder, cc.balancerBuildOpts)
 	} else {
-		newBalancerName := cc.curBalancerName
-		// This is the first time handling resolved addresses, try to create a
-		// pickfirst balancer.
-		if newBalancerName == "" {
-			newBalancerName = pickfirstName
-		}
+		var isGRPCLB bool
 		for _, a := range addrs {
-			if a.Type == resolver.GRPCLB && newBalancerName != grpclbName {
-				// Switch from non-grpclb to grpclb.
-				newBalancerName = grpclbName
+			if a.Type == resolver.GRPCLB {
+				isGRPCLB = true
 				break
-			} else if a.Type == resolver.Backend && newBalancerName == grpclbName {
-				// Switch from grpclb to non-grpclb.
+			}
+		}
+		var newBalancerName string
+		if isGRPCLB {
+			newBalancerName = grpclbName
```

Comment:
Optional, but it's a go-ism:

```go
newBalancerName := grpclbName
if !isGRPCLB {
  ...
}
```

but it's a little awkward in this case, so either is probably fine.